File: src/main/g8/$service__packaged$/src/main/$java__packaged$/$servicePackage__packaged$/hello/HelloServiceAsyncImpl.java
package com.github.dapeng.hello;

import com.github.dapeng.core.SoaException;
import com.github.dapeng.hello.domain.Hello;
import com.github.dapeng.hello.service.HelloServiceAsync;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Future;

public class HelloServiceAsyncImpl implements HelloServiceAsync {
    @Override
    public Future<String> sayHello(Hello hello, long timeout) throws SoaException {
        return CompletableFuture.supplyAsync(() -> hello.toString());
    }
}

File: src/main/g8/$service__packaged$/src/main/$scala__packaged$/$servicePackage__packaged$/scala/hello/HelloServiceAsyncImpl.scala
package com.github.dapeng.scala.hello

import com.github.dapeng.hello.scala.domain.Hello
import com.github.dapeng.hello.scala.service.HelloServiceAsync
import scala.concurrent.ExecutionContext.Implicits.global

import scala.concurrent.Future

class HelloServiceAsyncImpl extends  HelloServiceAsync{
  /**
    *
    **/
  override def sayHello(hello: Hello, timeout: Long): Future[String] = {
    Future {s" hello: \${hello}, timeout: \${timeout} "}
  }
}